src/syscheckd/fim_db.c

```diff
@@ -190,14 +190,21 @@ fdb_t *fim_db_init(int memory) {
     if (error) {
         merror("SQL ERROR: %s", error);
         sqlite3_free(error);
-        return NULL;
+        goto free_fim;
```

Comment:
Here we need to call `fim_db_finalize_stmt(fdb_t *fim_sql)` to reset an finalize statements.